dialogue-core/src/main/java/com/palantir/dialogue/core/BalancedNodeSelectionStrategyChannel.java

```diff
@@ -161,25 +178,130 @@ public String toString() {
      * might change mid-sort, leading to undefined behaviour.
      */
-    static final class ChannelStats {
-        private final long inflight;
-        private final int recentFailures;
+    private static final class SortableChannel {
+        private final int score;
+        private final MutableChannelWithStats delegate;
 
-        @VisibleForTesting
-        final MutableChannelWithStats delegate;
-
-        ChannelStats(long inflight, int recentFailures, MutableChannelWithStats delegate) {
-            this.inflight = inflight;
-            this.recentFailures = recentFailures;
+        SortableChannel(int score, MutableChannelWithStats delegate) {
+            this.score = score;
             this.delegate = delegate;
         }
 
+        int getScore() {
+            return score;
+        }
+
         @Override
         public String toString() {
-            return "ChannelStats{inflight="
-                    + inflight + ", recentFailures="
-                    + recentFailures + ", delegate="
-                    + delegate + '}';
+            return "SortableChannel{score=" + score + ", delegate=" + delegate + '}';
+        }
+    }
+
+    private static void registerGauges(
+            TaggedMetricRegistry taggedMetrics, String channelName, ImmutableList<MutableChannelWithStats> channels) {
+        if (channels.size() > 10) {
+            log.info("Not registering gauges as there are too many nodes", SafeArg.of("count", channels.size()));
+            return;
+        }
+
+        for (int hostIndex = 0; hostIndex < channels.size(); hostIndex++) {
+            MetricName metricName = MetricName.builder()
+                    .safeName("dialogue.balanced.score")
+                    .putSafeTags("channel-name", channelName)
+                    .putSafeTags("hostIndex", Integer.toString(hostIndex))
+                    .build();
+            // Weak gauge ensures this object can be GCd. Itherwise the tagged metric registry could hold the last ref!
+            // Defensive averaging for the possibility that people create multiple channels with the same channelName.
+            DialogueInternalWeakReducingGauge.getOrCreate(
+                    taggedMetrics,
+                    metricName,
+                    c -> c.computeScore().getScore(),
+                    longStream -> {
+                        long[] longs = longStream.toArray();
+                        if (longs.length > 1) {
+                            log.warn(
+                                    "Multiple objects contribute to the same gauge, taking the average "
+                                            + "(beware this may be misleading)",
+                                    SafeArg.of("metricName", metricName),
+                                    SafeArg.of("values", Arrays.toString(longs)));
+                        }
+                        return Arrays.stream(longs).average().orElse(0);
```

Comment:
thoughts on `.orElse(-1)` ? Just occurred to me that we could in theory get into a state where these have all been GC'd? and then a score of 0 might seem really desirable?